contracts/GnosisSafe.sol

```diff
@@ -184,13 +190,14 @@ contract GnosisSafe
     * @param dataHash Hash of the data (could be either a message hash or transaction hash)
     * @param data That should be signed (this is passed to an external validator contract)
     * @param signatures Signature data that should be verified. Can be ECDSA signature, contract signature (EIP-1271) or approved hash.
-    * @param consumeHash Indicates that in case of an approved hash the storage can be freed to save gas
```

Comment:
Agree, creating a new contract and deleting it for every transaction that requires on-chain approval requires more gas than in the current implementation. I think it can only be more efficient in the case of having a lot of owners signing.

Reply:
Yes, I agree that the current PR is not gas efficient, its was designed to allow the cleanup in a single call. 
If there is no problem in leaving junk in state, than we could just remove this consumeHash stuff. It's sad that we dont have a better way of doing this.